### PR TITLE
TINY-9843: Fix border styles being split into longhand form after table updates

### PR DESCRIPTION
**Pull request by tiny-ben-tran**
Related Ticket: TINY-9843

Description of Changes:
* Fix border styles in 'data-mce-style` being split into the longhand form after table updates

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/`, `hotfix/` or `spike/`

Review:
* [x] Milestone set
* [x] Docs ticket created (if applicable)

GitHub issues (if applicable):

### CODE REVIEW

**Comment by tiny-ben-tran** [comment]
- [] wait till after 6.5 is released

File: modules/tinymce/src/core/main/ts/api/dom/DOMUtils.ts [hunk 1]
```diff
@@ -286,6 +286,8 @@ const setupAttrHooks = (styles: Styles, settings: Partial<DOMUtilsSettings>, get
     style: {
       set: (elm, value) => {
         const sugarElm = SugarElement.fromDom(elm);
+        // TINY-9838: Prevent converting styles into shorthands for table element
+        const compressed = elm.nodeName !== 'TABLE';
```
Comment (MitchCTiny) [inline modules/tinymce/src/core/main/ts/api/dom/DOMUtils.ts:290]:
```suggestion
        const shouldCompress = elm.nodeName !== 'TABLE';
```
Maybe a variable name which describes its type and purpose?

File: modules/tinymce/src/core/main/ts/api/html/Styles.ts [hunk 2]
```diff
@@ -268,13 +269,15 @@ const Styles = (settings: StylesSettings = {}, schema?: Schema): Styles => {
...
         }
         // Compress the styles to reduce it's size for example IE will expand styles
-        compress('border', '', true);
-        compress('border', '-width');
-        compress('border', '-color');
-        compress('border', '-style');
-        compress('padding', '');
-        compress('margin', '');
-        compress2('border', 'border-width', 'border-style', 'border-color');
+        if (compressed) {
```
Comment (MitchCTiny) [inline modules/tinymce/src/core/main/ts/api/html/Styles.ts:272]:
```suggestion
        if (shouldCompress) {
```
Again, compressed implies already compressed, so rename for clarity.

**Review by MitchCTiny — APPROVED** [review]

**Review by mkzhx — APPROVED** [review]

File: modules/tinymce/CHANGELOG.md [hunk 3]
```diff
@@ -38,9 +38,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - If the selection contains more than one table cell, Quickbar toolbars are now positioned in the middle of the selection horizontally. #TINY-8297
 - Exposed `dataTransfer` property of drag and drop events for elements with a `contenteditable="false"` attribute. #TINY-9601
 - Screen readers now announce instructions for resizing the editor using arrow keys, when the resize handle is focused. #TINY-9793
-- Dialog `tabpanel` tab labels are now allowed to word wrap for better readability with long labels. #TINY-9947
+- Dialog `tabpanel` tab labels are now allowed to word wrap for better readability with long labels #TINY-9947
 - Added newlines before and after `details` elements in the output HTML. #TINY-9959
 - Added padding for empty `summary` elements so that they can be properly edited. #TINY-9959
+- The `parse()` of `tinymce.html.Styles` API now accepts an optional `compressed` parameter to disable converting styles into shorthand form. #TINY-9843
```
Comment (mkzhx) [inline modules/tinymce/CHANGELOG.md:44]:
```suggestion
- `tinymce.html.Styles.parse()` API now accepts an optional `shouldCompress` parameter to disable converting styles into shorthand form. #TINY-9843
```

File: modules/tinymce/CHANGELOG.md [hunk 4]
```diff
@@ -92,6 +93,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Pressing the Backspace or Delete key when the insertion point was within a `details` element resulted in broken markup. #TINY-9884
 - Making the selection into a list did not work if the selection included a block element with a `contenteditable="false"` attribute. #TINY-9823
 - Inserting elements in the middle of the summary caused two summaries to appear within `details` elements. #TINY-9885
+- The border style of table elements were optimized into a shorthand form after updated through the dialog. #TINY-9843
```
Comment (mkzhx) [inline modules/tinymce/CHANGELOG.md:96]:
```suggestion
- The `border` style of table elements were optimized into shorthand form after being updated through the Table Properties dialog. #TINY-9843
```

**Review by mkzhx — APPROVED** [review]

File: modules/tinymce/src/core/test/ts/browser/html/StylesTest.ts [hunk 5]
```diff
@@ -144,6 +144,84 @@ describe('browser.tinymce.core.html.StylesTest', () => {
     );
   });
 
+  it('TINY-9843: Should not compress styles', () => {
+    const styles = Styles();
+
+    assert.equal(
```
Comment (mkzhx) [inline modules/tinymce/src/core/test/ts/browser/html/StylesTest.ts:150]: Creating a function to abstract this `assert.equal(styles.serialize(styles.parse()))` code pattern would probably be good.

File: modules/tinymce/src/core/main/ts/api/html/Styles.ts [hunk 6]
```diff
@@ -31,7 +31,7 @@ export interface StylesSettings {
 }
 
 interface Styles {
-  parse: (css: string | undefined) => Record<string, string>;
+  parse: (css: string | undefined, compressed?: boolean) => Record<string, string>;
```
Comment (mkzhx) [inline modules/tinymce/src/core/main/ts/api/html/Styles.ts:34]:
```suggestion
  parse: (css: string | undefined, shouldCompress?: boolean) => Record<string, string>;
```

**Review by mkzhx — APPROVED** [review]

**Review by MitchCTiny — APPROVED** [review]

File: modules/tinymce/CHANGELOG.md [hunk 7]
```diff
@@ -13,6 +13,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Fixed
 - Returning an empty string in custom context menu update function would result in a small white line appearing on right click and the native browser context menu would not show up. #TINY-9842
 - Tab navigation incorrectly stopped around `iframe` dialog component. #TINY-9815
+- The border style of table was splitted into a longhand form after table updates, resulting a difference when being emitted during `getContent()`. #TINY-9843
```
Comment (ltrouton) [inline modules/tinymce/CHANGELOG.md:16]: I think we can keep it simple
```suggestion
- The border styles of a table were incorrectly split into a longhand form after table dialog updates. #TINY-9843
```

**Review by ltrouton — APPROVED** [review]

File: modules/tinymce/src/plugins/table/test/ts/browser/ui/TableDialogTest.ts [hunk 8]
```diff
@@ -489,14 +515,31 @@ describe('browser.tinymce.plugins.table.TableDialogTest', () => {
     };
 
     it('TINY-9837: Should not apply border, cellpadding or bordercolor data to cells if none of them has been modified',
-      () => testApplyDataToCells({ caption: true }, baseHtml));
+      () => {
+        const expectedHtml = [
```
[the remark this answers is not in the document]
Comment (tiny-ben-tran) [reply modules/tinymce/src/plugins/table/test/ts/browser/ui/TableDialogTest.ts]: I updated `baseHtml` so the test could be reverted.

File: modules/tinymce/src/core/test/ts/browser/html/StylesTest.ts (hunk 5 above)
Comment (tiny-ben-tran) [reply modules/tinymce/src/core/test/ts/browser/html/StylesTest.ts]: Done

**Review by spocke — APPROVED** [review]

File: modules/tinymce/src/plugins/table/test/ts/browser/ui/TableDialogTest.ts [hunk 9]
```diff
@@ -573,10 +599,10 @@ describe('browser.tinymce.plugins.table.TableDialogTest', () => {
 
     it('TINY-9837: Should apply cellpadding and bordercolor to cells if they have been modified',
       () => testApplyDataToCells({ cellpadding: '3px', bordercolor: '#FF0000' }, [
-        '<table style="border-collapse: collapse; width: 500px; height: 500px; border-width: 2px; border-color: #FF0000; border-style: dotted;" border="1"><colgroup><col style="width: 50%;"><col style="width: 50%;"></colgroup>',
+        '<table style="width: 500px; height: 500px; border: 2px dotted rgb(255, 0, 0); border-collapse: collapse;" border="1"><colgroup><col style="width: 50%;"><col style="width: 50%;"></colgroup>',
```
Comment (spocke) [inline modules/tinymce/src/plugins/table/test/ts/browser/ui/TableDialogTest.ts:602]: Why was this changed from hex to RGB?

Comment (tiny-ben-tran) [reply modules/tinymce/src/plugins/table/test/ts/browser/ui/TableDialogTest.ts:602]: Because [the new change](https://github.com/tinymce/tinymce/pull/8813/files#diff-278a6644e0ffb99813d0019eca11c186c502815d373ae299b603d6a10f208471L99) is to update the dom style instead of updating the style attribute. As a side effect of [dom.style.setProperty()](https://github.com/tinymce/tinymce/blob/2d8b95517a43d1f1967a2f3898371865a2fa18e4/modules/sugar/src/main/ts/ephox/sugar/api/properties/Css.ts#L21), hex is converted to rgb
